### PR TITLE
Fixed issue #18401: Unable to use checkdate function if debug set 

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

**Review by Shnoulle — APPROVED** [review]

**Comment by gabrieljenik** [comment]
Pnding of hooking new function to expression manager

File: tests/unit/helpers/CheckDateWrapTest.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     * Testing that exprmgr_checkdate evaluates
+     * the year parameter correctly.
+     *
+     * The year parameter must be an integer value
+     * or an integer numeric string.
+     */
+    public function testYearCheck()
+    {
+        $check_string = exprmgr_checkdate(06, 07, 'Twenty Twenty-three');
+        $this->assertFalse($check_string, 'Unexpected evaluation result, the year parameter should not be a string.');
```
Comment (Shnoulle) [inline tests/unit/helpers/CheckDateWrapTest.php:55]: :)

File: tests/unit/helpers/CheckDateWrapTest.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        $check_numeric = exprmgr_checkdate(06, 07, '2023');
+        $this->assertTrue($check_numeric, 'Unexpected evaluation result, the year parameter can be a numeric value.');
```
Comment (Shnoulle) [inline tests/unit/helpers/CheckDateWrapTest.php:58]: :smile:

File: tests/unit/helpers/CheckDateWrapTest.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+        $check_int = exprmgr_checkdate(06, 07, 2023);
+        $this->assertTrue($check_int, 'Unexpected evaluation result, the year parameter can be an integer.');
```
Comment (Shnoulle) [inline tests/unit/helpers/CheckDateWrapTest.php:61]: :+1:

**Review by Shnoulle — APPROVED** [review]
OK

Maybe adding some invalid date checkdate(2,31,2021)  checkdate(13,13,2021)

**Pull request merged**